### PR TITLE
rlp: pool decode streams

### PR DESCRIPTION
**Pull request by MariusVanDerWijden**
This pr introduces a pool of Stream objects to eliminate the need to reallocate memory for every new Stream. It should help to improve the performance of every call to Decode and Decode bytes. 

New streams will be allocated whenever needed.

### CODE REVIEW

**Comment by fjl** [comment]
Do you have benchmark results showing the improvement?

**Comment by fjl** [comment]
The pool in this PR is not safe for concurrent use.

**Comment by MariusVanDerWijden** [comment]
Yes I know, I've rewritten it, will update shortly

**Comment by MariusVanDerWijden** [comment]
@fjl can you review again? 
It significantly drops the time and amount of memory allocated for repeated calls to web3.getBlock()

**Comment by MariusVanDerWijden** [comment]
Anyone? @fjl @holiman @karalabe 
It's the same pattern used in log, swarm and trie

 https://github.com/ethereum/go-ethereum/blob/a9835c1816bc49ee54c82b4f2a5b05cbcd89881b/log/format.go#L360

File: rlp/decode.go [hunk 1]
```diff
@@ -48,6 +49,14 @@ var (
 	errUintOverflow  = errors.New("rlp: uint overflow")
 	errNoPointer     = errors.New("rlp: interface given to Decode must be a pointer")
 	errDecodeIntoNil = errors.New("rlp: pointer given to Decode must not be nil")
+
+	//Stream pool
```
Comment (karalabe) [inline rlp/decode.go]: I don't think you need this comment here, it's not providing any more info than the variable name.

File: rlp/decode.go [hunk 2]
```diff
@@ -48,6 +49,14 @@ var (
...
+	streamPool = sync.Pool{
+		//New is called whenever the pool is empty
```
Comment (karalabe) [inline rlp/decode.go]: No need to comment New, it's a struct field that has a documentation attached.

File: rlp/decode.go [hunk 3]
```diff
@@ -846,13 +862,14 @@ func (s *Stream) Reset(r io.Reader, inputLimit uint64) {
 	}
 	s.r = bufr
 	// Reset the decoding context.
-	s.stack = s.stack[:0]
+	s.stack = s.stack[:0:0]
```
Comment (karalabe) [inline rlp/decode.go]: What's the purpose of this change?

File: rlp/decode.go [hunk 4]
```diff
@@ -126,17 +135,24 @@ type Decoder interface {
...
 // Please see the documentation of Decode for the decoding rules.
 // The input must contain exactly one value and no trailing data.
 func DecodeBytes(b []byte, val interface{}) error {
-	// TODO: this could use a Stream from a pool.
 	r := bytes.NewReader(b)
-	if err := NewStream(r, uint64(len(b))).Decode(val); err != nil {
+
+	stream := streamPool.Get().(*Stream)
+	defer streamPool.Put(stream)
+	stream.Reset(r, uint64(len(b)))
```
Comment (karalabe) [inline rlp/decode.go]: Please leave an empty line before this one, and no empty line after. Code gets cleaner.

**Comment by MariusVanDerWijden** [comment]
# Benchmarks
I've written a script that queries geth for the first 72000 blocks. Each output represents 3000 blocks.

Important takeaways: (new vs. old)
85 vs. 91 seconds for the whole script
Total allocations: 4484791080 vs 5235291552
Mallocs: 37479991 vs. 39315777

## Original: (Geth/v1.9.0-unstable-f1537b77/linux-amd64/go1.11.5)
block number:72000
percent parsed: 0.000 elapsed time: 1s
percent parsed: 0.042 elapsed time: 6s 
percent parsed: 0.083 elapsed time: 10s
percent parsed: 0.125 elapsed time: 13s
percent parsed: 0.167 elapsed time: 17s
percent parsed: 0.208 elapsed time: 20s 
percent parsed: 0.250 elapsed time: 23s 
percent parsed: 0.292 elapsed time: 27s 
percent parsed: 0.333 elapsed time: 31s 
percent parsed: 0.375 elapsed time: 35s
percent parsed: 0.417 elapsed time: 39s
percent parsed: 0.458 elapsed time: 43s
percent parsed: 0.500 elapsed time: 47s
percent parsed: 0.542 elapsed time: 51s
percent parsed: 0.583 elapsed time: 55s
percent parsed: 0.625 elapsed time: 58s
percent parsed: 0.667 elapsed time: 62s
percent parsed: 0.708 elapsed time: 66s 
percent parsed: 0.750 elapsed time: 69s
percent parsed: 0.792 elapsed time: 73s
percent parsed: 0.833 elapsed time: 77s
percent parsed: 0.875 elapsed time: 81s
percent parsed: 0.917 elapsed time: 84s
percent parsed: 0.958 elapsed time: 88s 
Generated in 91 seconds

```
debug.memStats()
{
  Alloc: 792948520,
  BuckHashSys: 1772804,
  BySize: [{
      Frees: 0,
      Mallocs: 0,
      Size: 0
  }, {
      Frees: 2187187,
      Mallocs: 2245550,
      Size: 8
  }, {
      Frees: 10632025,
      Mallocs: 10683766,
      Size: 16
  }, {
      Frees: 8305248,
      Mallocs: 8391877,
      Size: 32
  }, {
      Frees: 5412904,
      Mallocs: 5756974,
      Size: 48
  }, {
      Frees: 2579259,
      Mallocs: 2875407,
      Size: 64
  }, {
      Frees: 782822,
      Mallocs: 871319,
      Size: 80
  }, {
      Frees: 534477,
      Mallocs: 556562,
      Size: 96
  }, {
      Frees: 431589,
      Mallocs: 474285,
      Size: 112
  }, {
      Frees: 339237,
      Mallocs: 346696,
      Size: 128
  }, {
      Frees: 316059,
      Mallocs: 317905,
      Size: 144
  }, {
      Frees: 150001,
      Mallocs: 150824,
      Size: 160
  }, {
      Frees: 2830,
      Mallocs: 3637,
      Size: 176
  }, {
      Frees: 1933,
      Mallocs: 2107,
      Size: 192
  }, {
      Frees: 3610,
      Mallocs: 17820,
      Size: 208
  }, {
      Frees: 73028,
      Mallocs: 73115,
      Size: 224
  }, {
      Frees: 1618491,
      Mallocs: 1782389,
      Size: 240
  }, {
      Frees: 153886,
      Mallocs: 156052,
      Size: 256
  }, {
      Frees: 14393,
      Mallocs: 15799,
      Size: 288
  }, {
      Frees: 2606,
      Mallocs: 2827,
      Size: 320
  }, {
      Frees: 366574,
      Mallocs: 371137,
      Size: 352
  }, {
      Frees: 3084,
      Mallocs: 3812,
      Size: 384
  }, {
      Frees: 5309,
      Mallocs: 6466,
      Size: 416
  }, {
      Frees: 194560,
      Mallocs: 196439,
      Size: 448
  }, {
      Frees: 73186,
      Mallocs: 73333,
      Size: 480
  }, {
      Frees: 153547,
      Mallocs: 154421,
      Size: 512
  }, {
      Frees: 473831,
      Mallocs: 475824,
      Size: 576
  }, {
      Frees: 2027,
      Mallocs: 2440,
      Size: 640
  }, {
      Frees: 5742,
      Mallocs: 6026,
      Size: 704
  }, {
      Frees: 3244,
      Mallocs: 3724,
      Size: 768
  }, {
      Frees: 74674,
      Mallocs: 75464,
      Size: 896
  }, {
      Frees: 3931,
      Mallocs: 4964,
      Size: 1024
  }, {
      Frees: 73171,
      Mallocs: 73316,
      Size: 1152
  }, {
      Frees: 1580,
      Mallocs: 1866,
      Size: 1280
  }, {
      Frees: 4765,
      Mallocs: 4910,
      Size: 1408
  }, {
      Frees: 67290,
      Mallocs: 67370,
      Size: 1536
  }, {
      Frees: 4101,
      Mallocs: 4533,
      Size: 1792
  }, {
      Frees: 3786,
      Mallocs: 4554,
      Size: 2048
  }, {
      Frees: 644,
      Mallocs: 755,
      Size: 2304
  }, {
      Frees: 1668,
      Mallocs: 1982,
      Size: 2688
  }, {
      Frees: 1117,
      Mallocs: 1254,
      Size: 3072
  }, {
      Frees: 275,
      Mallocs: 306,
      Size: 3200
  }, {
      Frees: 1420,
      Mallocs: 1749,
      Size: 3456
  }, {
      Frees: 3825,
      Mallocs: 4620,
      Size: 4096
  }, {
      Frees: 7850,
      Mallocs: 10729,
      Size: 4864
  }, {
      Frees: 9171,
      Mallocs: 10791,
      Size: 5376
  }, {
      Frees: 3424,
      Mallocs: 3811,
      Size: 6144
  }, {
      Frees: 471,
      Mallocs: 553,
      Size: 6528
  }, {
      Frees: 694,
      Mallocs: 828,
      Size: 6784
  }, {
      Frees: 416,
      Mallocs: 548,
      Size: 6912
  }, {
      Frees: 2658,
      Mallocs: 4078,
      Size: 8192
  }, {
      Frees: 25535,
      Mallocs: 45789,
      Size: 9472
  }, {
      Frees: 405,
      Mallocs: 494,
      Size: 9728
  }, {
      Frees: 406,
      Mallocs: 510,
      Size: 10240
  }, {
      Frees: 443,
      Mallocs: 529,
      Size: 10880
  }, {
      Frees: 885,
      Mallocs: 1045,
      Size: 12288
  }, {
      Frees: 852,
      Mallocs: 1042,
      Size: 13568
  }, {
      Frees: 950,
      Mallocs: 1202,
      Size: 14336
  }, {
      Frees: 8754,
      Mallocs: 10480,
      Size: 16384
  }, {
      Frees: 1676,
      Mallocs: 2023,
      Size: 18432
  }, {
      Frees: 478,
      Mallocs: 565,
      Size: 19072
  }],
  DebugGC: false,
  EnableGC: true,
  Frees: 38080739,
  GCCPUFraction: 0.00033411981545787957,
  GCSys: 49516544,
  HeapAlloc: 792948520,
  HeapIdle: 383983616,
  HeapInuse: 888889344,
  HeapObjects: 1235038,
  HeapReleased: 0,
  HeapSys: 1272872960,
  LastGC: 1550580403260897800,
  Lookups: 0,
  MCacheInuse: 27648,
  MCacheSys: 32768,
  MSpanInuse: 6353752,
  MSpanSys: 10977280,
  Mallocs: 39315777,
  NextGC: 1269062240,
  NumForcedGC: 0,
  NumGC: 24,
  OtherSys: 4505588,
  PauseEnd: [1550580258812800300, 1550580258818635800, 1550580258845799400, 1550580258883930600, 1550580258905833700, 1550580258916076500, 1550580258931174700, 1550580258961192200, 1550580258984503000, 1550580259029328400, 1550580259169294800, 1550580259324328700, 1550580259398989300, 1550580259447528400, 1550580259656058600, 1550580307045286400, 1550580308308521000, 1550580317020484000, 1550580327357460700, 1550580338525496800, 1550580355289617700, 1550580373694388500, 1550580389065160700, 1550580403260897800, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0],
  PauseNs: [148998, 150854, 131610, 148126, 140115, 137237, 140069, 153378, 146728, 148572, 131372, 93861, 1502285, 141288, 100994, 295332, 255802, 164740, 122638, 160890, 173659, 196386, 155076, 125946, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0],
  PauseTotalNs: 5065956,
  StackInuse: 2195456,
  StackSys: 2195456,
  Sys: 1341873400,
  TotalAlloc: 5235291552
```

## This PR 

block number:72000
percent parsed: 0.000 elapsed time: 1s memoryUsage: 19.96MB
percent parsed: 0.042 elapsed time: 6s memoryUsage: 37.64MB
percent parsed: 0.083 elapsed time: 10s memoryUsage: 30.18MB
percent parsed: 0.125 elapsed time: 13s memoryUsage: 22.07MB
percent parsed: 0.167 elapsed time: 16s memoryUsage: 26.37MB
percent parsed: 0.208 elapsed time: 23s memoryUsage: 30.05MB
percent parsed: 0.250 elapsed time: 26s memoryUsage: 22.02MB
percent parsed: 0.292 elapsed time: 29s memoryUsage: 27.44MB
percent parsed: 0.333 elapsed time: 32s memoryUsage: 19.52MB
percent parsed: 0.375 elapsed time: 35s memoryUsage: 34.53MB
percent parsed: 0.417 elapsed time: 38s memoryUsage: 26.49MB
percent parsed: 0.458 elapsed time: 41s memoryUsage: 30.23MB
percent parsed: 0.500 elapsed time: 45s memoryUsage: 22.09MB
percent parsed: 0.542 elapsed time: 48s memoryUsage: 25.89MB
percent parsed: 0.583 elapsed time: 51s memoryUsage: 29.28MB
percent parsed: 0.625 elapsed time: 55s memoryUsage: 20.88MB
percent parsed: 0.667 elapsed time: 58s memoryUsage: 26.12MB
percent parsed: 0.708 elapsed time: 62s memoryUsage: 18.13MB
percent parsed: 0.750 elapsed time: 65s memoryUsage: 33.24MB
percent parsed: 0.792 elapsed time: 69s memoryUsage: 25.24MB
percent parsed: 0.833 elapsed time: 72s memoryUsage: 28.65MB
percent parsed: 0.875 elapsed time: 76s memoryUsage: 32.22MB
percent parsed: 0.917 elapsed time: 79s memoryUsage: 24.15MB
percent parsed: 0.958 elapsed time: 83s memoryUsage: 28.40MB
Generated in 85 seconds

```
debug.memStats()
{
  Alloc: 1045778152,
  BuckHashSys: 1743891,
  BySize: [{
      Frees: 0,
      Mallocs: 0,
      Size: 0
  }, {
      Frees: 2063686,
      Mallocs: 2364499,
      Size: 8
  }, {
      Frees: 9522842,
      Mallocs: 10473858,
      Size: 16
  }, {
      Frees: 7520463,
      Mallocs: 8453530,
      Size: 32
  }, {
      Frees: 4598437,
      Mallocs: 5348832,
      Size: 48
  }, {
      Frees: 1915025,
      Mallocs: 2345476,
      Size: 64
  }, {
      Frees: 702176,
      Mallocs: 865030,
      Size: 80
  }, {
      Frees: 478805,
      Mallocs: 551118,
      Size: 96
  }, {
      Frees: 413642,
      Mallocs: 507482,
      Size: 112
  }, {
      Frees: 91732,
      Mallocs: 103417,
      Size: 128
  }, {
      Frees: 278963,
      Mallocs: 315144,
      Size: 144
  }, {
      Frees: 134886,
      Mallocs: 149276,
      Size: 160
  }, {
      Frees: 1331,
      Mallocs: 2037,
      Size: 176
  }, {
      Frees: 1650,
      Mallocs: 1919,
      Size: 192
  }, {
      Frees: 1921,
      Mallocs: 19442,
      Size: 208
  }, {
      Frees: 65787,
      Mallocs: 72858,
      Size: 224
  }, {
      Frees: 1078171,
      Mallocs: 1297652,
      Size: 240
  }, {
      Frees: 136616,
      Mallocs: 153380,
      Size: 256
  }, {
      Frees: 12063,
      Mallocs: 14510,
      Size: 288
  }, {
      Frees: 1805,
      Mallocs: 2010,
      Size: 320
  }, {
      Frees: 331662,
      Mallocs: 370374,
      Size: 352
  }, {
      Frees: 1248,
      Mallocs: 1817,
      Size: 384
  }, {
      Frees: 3627,
      Mallocs: 4670,
      Size: 416
  }, {
      Frees: 165355,
      Mallocs: 192047,
      Size: 448
  }, {
      Frees: 65809,
      Mallocs: 72795,
      Size: 480
  }, {
      Frees: 135223,
      Mallocs: 150821,
      Size: 512
  }, {
      Frees: 430035,
      Mallocs: 474387,
      Size: 576
  }, {
      Frees: 1547,
      Mallocs: 2008,
      Size: 640
  }, {
      Frees: 5303,
      Mallocs: 5802,
      Size: 704
  }, {
      Frees: 2222,
      Mallocs: 2878,
      Size: 768
  }, {
      Frees: 67458,
      Mallocs: 75944,
      Size: 896
  }, {
      Frees: 2137,
      Mallocs: 3695,
      Size: 1024
  }, {
      Frees: 65855,
      Mallocs: 72844,
      Size: 1152
  }, {
      Frees: 834,
      Mallocs: 1056,
      Size: 1280
  }, {
      Frees: 2954,
      Mallocs: 4133,
      Size: 1408
  }, {
      Frees: 62037,
      Mallocs: 67258,
      Size: 1536
  }, {
      Frees: 2536,
      Mallocs: 3398,
      Size: 1792
  }, {
      Frees: 1621,
      Mallocs: 2404,
      Size: 2048
  }, {
      Frees: 317,
      Mallocs: 458,
      Size: 2304
  }, {
      Frees: 603,
      Mallocs: 925,
      Size: 2688
  }, {
      Frees: 526,
      Mallocs: 706,
      Size: 3072
  }, {
      Frees: 137,
      Mallocs: 231,
      Size: 3200
  }, {
      Frees: 558,
      Mallocs: 812,
      Size: 3456
  }, {
      Frees: 1657,
      Mallocs: 2362,
      Size: 4096
  }, {
      Frees: 12211,
      Mallocs: 15286,
      Size: 4864
  }, {
      Frees: 4003,
      Mallocs: 5272,
      Size: 5376
  }, {
      Frees: 1995,
      Mallocs: 2320,
      Size: 6144
  }, {
      Frees: 185,
      Mallocs: 255,
      Size: 6528
  }, {
      Frees: 272,
      Mallocs: 393,
      Size: 6784
  }, {
      Frees: 170,
      Mallocs: 258,
      Size: 6912
  }, {
      Frees: 40181,
      Mallocs: 55429,
      Size: 8192
  }, {
      Frees: 489,
      Mallocs: 707,
      Size: 9472
  }, {
      Frees: 171,
      Mallocs: 254,
      Size: 9728
  }, {
      Frees: 156,
      Mallocs: 235,
      Size: 10240
  }, {
      Frees: 151,
      Mallocs: 213,
      Size: 10880
  }, {
      Frees: 289,
      Mallocs: 441,
      Size: 12288
  }, {
      Frees: 562,
      Mallocs: 778,
      Size: 13568
  }, {
      Frees: 333,
      Mallocs: 526,
      Size: 14336
  }, {
      Frees: 5497,
      Mallocs: 9987,
      Size: 16384
  }, {
      Frees: 488,
      Mallocs: 776,
      Size: 18432
  }, {
      Frees: 141,
      Mallocs: 212,
      Size: 19072
  }],
  DebugGC: false,
  EnableGC: true,
  Frees: 33263453,
  GCCPUFraction: 0.0006056832344229216,
  GCSys: 49225728,
  HeapAlloc: 1045778152,
  HeapIdle: 207282176,
  HeapInuse: 1065525248,
  HeapObjects: 4216538,
  HeapReleased: 0,
  HeapSys: 1272807424,
  LastGC: 1550580606556649700,
  Lookups: 0,
  MCacheInuse: 27648,
  MCacheSys: 32768,
  MSpanInuse: 10465656,
  MSpanSys: 12861440,
  Mallocs: 37479991,
  NextGC: 1248354688,
  NumForcedGC: 0,
  NumGC: 23,
  OtherSys: 4841701,
  PauseEnd: [1550580518579803600, 1550580518584882200, 1550580518611261400, 1550580518650297300, 1550580518675692000, 1550580518691087600, 1550580518721246200, 1550580518754974000, 1550580518796748500, 1550580518852380200, 1550580519065391800, 1550580519105447700, 1550580519119885300, 1550580519163980500, 1550580519387568000, 1550580523546469000, 1550580535056590600, 1550580543383162000, 1550580559744737500, 1550580570372601900, 1550580579867437300, 1550580593558388700, 1550580606556649700, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0],
  PauseNs: [113449, 137431, 150330, 147220, 151549, 150628, 165551, 158459, 153584, 136273, 170289, 133366, 137770, 10365162, 103230, 181302, 210639, 949043, 158636, 133614, 232409, 157009, 150713, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0],
  PauseTotalNs: 14547656,
  StackInuse: 2260992,
  StackSys: 2260992,
  Sys: 1343773944,
  TotalAlloc: 4484791080
```

File: rlp/decode.go (hunk 1 above)
Comment (MariusVanDerWijden) [reply rlp/decode.go]: fixed
